crates/ide/src/references.rs

```diff
@@ -2139,4 +2145,28 @@ fn test() {
             "#]],
         );
     }
+
+    #[test]
+    fn multiple_references_inside_macro() {
+        check(
+            r#"
+macro_rules! use_name {
+    ($name:ident) => {
+        let _ = $name;
+        let _ = $name;
+        let _ = $name;
+    };
+}
+fn test() {
+    let name = ();
+    use_name!(name$0);
```

Comment:
> Apparently our descend_into_macros() stanza returns one definition for
> each time a name is used in a macro.

Sorry this was misleading. It's not about definitions, only references.

Given

```rust
macro_rules! use_name {
    ($name:ident) => {
        println!("{}", $name);
        println!("{}", $name);
        println!("{}", $name);
    };
}
fn test() {
    let name = 123;
    use_name!(name);
    println!("{}", name);
    println!("{}", name);
}
```

so 3 references inside the macro, and 4 outside;
this will produce 12 references; it's the product. Not sure if this is expected but I guess I haven't seen another problem with it.

It only surfaced when there was a definition inside a macro because we already deduplicate the per-definition references.

I think your suggestion makes sense; I'll try it and maybe get rid of the other two `unique()` calls.